# iNaturalist status updates by state

Using the file produced in the collate-status-taxa.ipynb: `inat-aust-status-taxa.csv`, generate lists to update iNaturalist statuses

**Next steps:**
State by state establish the changes that need to be made:
    a. new - any new species that appear in the state lists but do not have a status in inaturalist (new template)
    b. updates - any changes to information which was added by us previously (user_id = 708886) (update template, action='UPDATE')
    c. removals - any statuses which were added by us previously (user_id = 708886) list which are incorrect (update template, action='REMOVE')
    d. flags - are there any statuses by other users that need to be flagged?

## Prep - common to all states
1. Read in the inaturalist statuses & filter out this state
2. Read in the inaturalist taxa list
3. Read in the state sensitive list
4. Attempt to match the state statuses to an IUCN equivalent


### 1. iNaturalist statuses

In [85]:
import pandas as pd

projectdir = "/Users/oco115/PycharmProjects/authoritative-lists/" # basedir for this gh project
# projectdir = "/Users/new330/IdeaProjects/authoritative-lists/" # basedir for this gh project
sourcedir = projectdir + "source-data/inaturalist-statuses/"
listdir = projectdir + "current-lists/"


# read in the statuses
taxastatus = pd.read_csv(sourcedir + "inat-aust-status-taxa.csv", encoding='UTF-8',na_filter=False,dtype=str) ## Read inaturalist conservation statuses file
# taxastatus.head(3)
taxastatus

id taxon_id  user_id place_id source_id  \
0     166449    38493  1138587     7830             
1     234788   918383   702203     9994             
2     234789   918383   702203     7308             
3     166416  1033183  3669610     6825             
4     180721  1247288   222137     6825             
...      ...      ...      ...      ...       ...   
3448  165697  1182117  3669610    73684             
3449  130966   508985                               
3450  161226    50744   516268                      
3451  162783   924263   764897                      
3452  152695   779693   708886              16653   

                                              authority  \
0                    Flora and Fauna Guarantee Act 1988   
1                             Atlas of Living Australia   
2                             Atlas of Living Australia   
3                  NSW Office of Environment & Heritage   
4           NSW Threatened Species Scientific Committee   
...                                                 ...   
3448                              Australian Government   
3449                                                      
3450          DAWE Species Profile and Threats Database   
3451  Department of Biodiversity, Conservation and A...   
3452                                           QLD DEHP   

                     status  \
0                        CR   
1                       NT    
2                        LC   
3                        EN   
4                        vu   
...                     ...   
3448                     CR   
3449  critically endangered   
3450                     CR   
3451                     EN   
3452             endangered   

                                                    url description  \
0                                                                     
1     https://bie.ala.org.au/species/https://id.biod...               
2     https://bie.ala.org.au/species/https://id.biod...               
3     https://www.environment.nsw.gov.au/threateneds...               
4     https://www.environment.nsw.gov.au/topics/anim...               
...                                                 ...         ...   
3448  http://www.environment.gov.au/biodiversity/thr...               
3449  http://environment.gov.au/cgi-bin/sprat/public...               
3450  https://www.environment.gov.au/cgi-bin/sprat/p...               
3451  https://www.dpaw.wa.gov.au/images/documents/pl...               
3452  https://data.qld.gov.au/dataset/conservation-s...               

     geoprivacy  ...          genus specificEpithet infraspecificEpithet  \
0      obscured  ...      Eulamprus      kosciuskoi                        
1                ...   Chiloschista     phyllorhiza                        
2                ...   Chiloschista     phyllorhiza                        
3      obscured  ...       Eidothea      hardeniana                        
4      obscured  ...     Pomaderris         bodalla                        
...         ...  ...            ...             ...                  ...   
3448             ...    Achyranthes    margaretarum                        
3449   obscured  ...  Lichenostomus        melanops             cassidix   
3450   obscured  ...                                                       
3451   obscured  ...         Reedia       spathacea                        
3452   obscured  ...   Selenocosmia       crassipes                        

                  modified                   scientificName   taxonRank  \
0     2021-03-01T10:35:01Z             Eulamprus kosciuskoi     species   
1     2022-01-08T03:30:36Z         Chiloschista phyllorhiza     species   
2     2022-01-08T03:30:36Z         Chiloschista phyllorhiza     species   
3     2021-02-22T07:21:17Z              Eidothea hardeniana     species   
4     2021-08-27T06:18:35Z               Pomaderris bodalla     species   
...                    ...                              ...         ...   


In [86]:
def filter_state_statuses(stateregex: str, urlregex: str):
    authoritydf = taxastatus['authority'].drop_duplicates().sort_values()
    authoritydf = authoritydf[pd.Series(authoritydf).str.contains(stateregex)]
    urldf = taxastatus['url'].drop_duplicates().sort_values()
    urldf = urldf[pd.Series(urldf).str.contains(urlregex)]
    placedisplaydf = taxastatus['place_display_name'].drop_duplicates().sort_values()
    placedisplaydf = placedisplaydf[pd.Series(placedisplaydf).str.contains(stateregex)]
    placedf = taxastatus['place_name'].drop_duplicates().sort_values()
    placedf = placedf[pd.Series(placedf).str.contains(stateregex)]
    # concat all and remove duplicates
    statedf = pd.concat([taxastatus.apply(lambda row: row[taxastatus['place_display_name'].isin(placedisplaydf)]),
                         taxastatus.apply(lambda row: row[taxastatus['place_name'].isin(placedf)]),
                         taxastatus.apply(lambda row: row[taxastatus['url'].isin(urldf)]),
                         taxastatus.apply(
                             lambda row: row[taxastatus['authority'].isin(authoritydf)])]).drop_duplicates()
    return statedf.sort_values(['taxon_id', 'user_id'])
inatstatuses = filter_state_statuses("NSW|New South Wales", ".nsw.")
inatstatuses.rename(columns={'id':'status_id','id_y':'taxon_id_y'},inplace=True)
inatstatuses

status_id taxon_id  user_id place_id source_id  \
3       166416  1033183  3669610     6825             
2656    165059  1054498    58320     6825             
477     264941  1061113  3669610     6825             
371     180988  1070573  3669610     6825             
2514    160581  1076814   990532     6825             
...        ...      ...      ...      ...       ...   
1111    152247   906670   708886     6825     16650   
2637    164342   913024    58320     6825             
7       167825   953250   702203     6825             
836     155144   966856   708886     6825     16650   
2651    164615   973457   222137     6825             

                                              authority      status  \
3                  NSW Office of Environment & Heritage          EN   
2656               NSW Office of Environment & Heritage  Vulnerable   
477   New South Wales Biodiversity Conservation Act ...          EX   
371                NSW Office of Environment & Heritage          EN   
2514               NSW Office of Environment & Heritage  Vulnerable   
...                                                 ...         ...   
1111               NSW Office of Environment & Heritage  Endangered   
2637               NSW Office of Environment & Heritage  Vulnerable   
7     New South Wales Office of Environment and Heri...          VU   
836                NSW Office of Environment & Heritage  endangered   
2651        NSW Threatened Species Scientific Committee  Endangered   

                                                    url       description  \
3     https://www.environment.nsw.gov.au/threateneds...                     
2656  https://www.environment.nsw.gov.au/threatenedS...                     
477   https://bie.ala.org.au/species/https://id.biod...  Presumed Extinct   
371   https://www.environment.nsw.gov.au/threateneds...                     
2514  https://www.environment.nsw.gov.au/threateneds...                     
...                                                 ...               ...   
1111  https://www.environment.nsw.gov.au/resources/a...                     
2637  https://www.environment.nsw.gov.au/threatenedS...                     
7     https://www.environment.nsw.gov.au/threatenedS...        Vulnerable   
836   https://www.environment.nsw.gov.au/resources/a...                     
2651  https://www.environment.nsw.gov.au/topics/anim...                     

     geoprivacy  ...         genus specificEpithet infraspecificEpithet  \
3      obscured  ...      Eidothea      hardeniana                        
2656   obscured  ...  Prostanthera  cryptandroides       cryptandroides   
477        open  ...        Leuzea       australis                        
371              ...        Rotala      tripartita                        
2514   obscured  ...   Pterostylis       nigricans                        
...         ...  ...           ...             ...                  ...   
1111   obscured  ...     Caladenia        concolor                        
2637             ...     Symplocos     baeuerlenii                        
7                ...     Pultenaea          glabra                        
836    obscured  ...        Pteris   platyzomopsis                        
2651   obscured  ...       Calotis       pubescens                        

                  modified                              scientificName  \
3     2021-02-22T07:21:17Z                         Eidothea hardeniana   
2656  2020-12-17T03:53:45Z  Prostanthera cryptandroides cryptandroides   
477   2022-06-11T11:38:00Z                            Leuzea australis   
371   2021-09-21T05:48:55Z                           Rotala tripartita   
2514  2022-07-12T14:19:22Z                       Pterostylis nigricans   
...                    ...                                         ...   
1111  2022-06-10T07:36:31Z                          Caladenia concolor   
2637  2022-03-14T15:27:03Z                       Symplocos baeuerlenii   

### 2. iNaturalist taxonomy

In [87]:
# Output files contain these fields
# Taxon_Name,Status,Authority,IUCN_equivalent,Description,iNaturalist_Place_ID,url,Taxon_Geoprivacy,Username,taxon_id
# so we need to match species from the state lists to the inat taxa to get the taxon_id

import zipfile
url = "https://www.inaturalist.org/taxa/inaturalist-taxonomy.dwca.zip"
filename = url.split("/")[-1]

z=zipfile.ZipFile(sourcedir + filename)

with z.open('taxa.csv') as from_archive:
    inattaxa = pd.read_csv(from_archive,dtype=str)
z.close()
inattaxa.head(3)


id                             taxonID                          identifier  \
0  1  https://www.inaturalist.org/taxa/1  https://www.inaturalist.org/taxa/1   
1  2  https://www.inaturalist.org/taxa/2  https://www.inaturalist.org/taxa/2   
2  3  https://www.inaturalist.org/taxa/3  https://www.inaturalist.org/taxa/3   

                         parentNameUsageID   kingdom    phylum class order  \
0   https://www.inaturalist.org/taxa/48460  Animalia       NaN   NaN   NaN   
1       https://www.inaturalist.org/taxa/1  Animalia  Chordata   NaN   NaN   
2  https://www.inaturalist.org/taxa/355675  Animalia  Chordata  Aves   NaN   

  family genus specificEpithet infraspecificEpithet              modified  \
0    NaN   NaN             NaN                  NaN  2021-11-02T06:05:44Z   
1    NaN   NaN             NaN                  NaN  2021-11-23T00:40:18Z   
2    NaN   NaN             NaN                  NaN  2022-12-27T07:33:16Z   

  scientificName taxonRank                                         references  
0       Animalia   kingdom  http://www.catalogueoflife.org/annual-checklis...  
1       Chordata    phylum  http://www.catalogueoflife.org/annual-checklis...  
2           Aves     class  http://www.catalogueoflife.org/annual-checklis...

In [ ]:
### 3. State lists

Sensitive list: `geoprivacy` = `obscured`
overrides


In [88]:
sensitivelist = pd.read_csv(listdir + "sensitive-lists/NSW-sensitive.csv")  # Qld sensitive list
sensitivelist['scientificName'] = sensitivelist['scientificName'].str.replace('subsp. ', '', regex=False)
sensitivelist = sensitivelist.rename(columns={'taxonID':'bionetTaxonID'})
sensitivelist['bionet_geoprivacy'] = 'obscured'
sensitivelist

taxonRank   kingdom     class           order       family  \
0       Species  Animalia      Aves  Psittaciformes   Cacatuidae   
1       Species  Animalia      Aves    Strigiformes    Strigidae   
2       Species  Animalia      Aves    Strigiformes    Tytonidae   
3       Species  Animalia      Aves    Strigiformes    Strigidae   
4       Species  Animalia  Reptilia        Squamata     Elapidae   
..          ...       ...       ...             ...          ...   
193  Subspecies   Plantae     Flora           Flora      Poaceae   
194     Species   Plantae     Flora           Flora  Orchidaceae   
195     Species   Plantae     Flora           Flora  Orchidaceae   
196     Species   Plantae     Flora           Flora  Orchidaceae   
197     Species   Plantae     Flora           Flora    Myrtaceae   

             genus                 scientificName specificEpithet  \
0    Callocephalon       Callocephalon fimbriatum      fimbriatum   
1            Ninox                  Ninox strenua         strenua   
2             Tyto               Tyto tenebricosa     tenebricosa   
3            Ninox                Ninox connivens       connivens   
4    Hoplocephalus      Hoplocephalus bungaroides     bungaroides   
..             ...                            ...             ...   
193    Anthosachne  Anthosachne kingiana kingiana        kingiana   
194      Caladenia                Caladenia tensa           tensa   
195      Caladenia           Caladenia atroclavia      atroclavia   
196   Prasophyllum           Prasophyllum sandrae         sandrae   
197     Backhousia         Backhousia subargentea     subargentea   

                  vernacularName    establishmentMeans  ... protectedInNSW  \
0             Gang-gang Cockatoo  Alive in NSW, Native  ...           True   
1                   Powerful Owl  Alive in NSW, Native  ...           True   
2                      Sooty Owl  Alive in NSW, Native  ...           True   
3                    Barking Owl  Alive in NSW, Native  ...           True   
4             Broad-headed Snake  Alive in NSW, Native  ...           True   
..                           ...                   ...  ...            ...   
193    Philip Island Wheat Grass  Alive in NSW, Native  ...          False   
194          Rigid Spider-orchid  Alive in NSW, Native  ...           True   
195  Black-clubbed Spider-orchid  Alive in NSW, Native  ...           True   
196     Majors Creek Leek Orchid  Alive in NSW, Native  ...          False   
197               Giant Ironwood  Alive in NSW, Native  ...          False   

     sensitivityClass tsprofileID    countryConservation  \
0          Category 3     10975.0             Endangered   
1          Category 3     10562.0             Not Listed   
2          Category 3     10821.0             Not Listed   
3          Category 3     10561.0             Not Listed   
4          Category 2     10413.0             Vulnerable   
..                ...         ...                    ...   
193        Category 3     20148.0  Critically Endangered   
194        Category 2         NaN             Not Listed   
195        Category 2     20365.0             Endangered   
196        Category 2     10668.0             Not Listed   
197        Category 3     10166.0             Not Listed   

                  dcterms_modified speciesID  bionetTaxonID  generalisation  \
0     2022-05-12T15:50:01.01+10:00         6              6             1km   
1        2011-11-11T11:23:06+11:00       162            162             1km   
2    2012-07-23T18:04:52.683+10:00       333            333             1km   
3        2011-11-11T11:23:06+11:00       363            363             1km   
4        2011-11-11T11:23:04+11:00       390            390            10km   
..                             ...       ...            ...             ...   
193   2020-01-23T17:14:13.01+11:00     23014          23014             1km   
194   2020-06-23T19:56:43.85+10:00     23737          23737           

In [89]:
def gpcondition(x):
    if x=='Category 1':
        return "private"
    elif x== 'Category 2' or x=='Category 3':
        return "obscured"
    else:
        return 'open'

In [90]:
conservationlist = pd.read_csv(listdir + "conservation-lists/NSW-conservation.csv")  # Qld sensitive list
conservationlist['scientificName'] = conservationlist['scientificName'].str.replace('subsp. ', '', regex=False)
conservationlist = conservationlist.rename(columns={'taxonID':'bionetTaxonID'})
conservationlist['bionet_geoprivacy'] = conservationlist['sensitivityClass'].apply(gpcondition)
conservationlist

taxonRank   kingdom     class           order          family  \
0        Species  Animalia  Reptilia        Squamata     Pygopodidae   
1        Species  Animalia      Aves  Psittaciformes      Cacatuidae   
2        Species  Animalia  Reptilia        Squamata        Elapidae   
3        Species  Animalia  Amphibia           Anura         Hylidae   
4        Species  Animalia      Aves   Passeriformes    Meliphagidae   
...          ...       ...       ...             ...             ...   
1017     Species   Plantae     Flora           Flora   Euphorbiaceae   
1018     Species   Plantae     Flora           Flora       Ericaceae   
1019     Species   Plantae     Flora           Flora  Pittosporaceae   
1020  Subspecies  Animalia      Aves   Passeriformes       Maluridae   
1021     Species  Animalia   Insecta      Coleoptera        Cleridae   

              genus                 scientificName    specificEpithet  \
0             Delma                    Delma impar              impar   
1     Callocephalon       Callocephalon fimbriatum         fimbriatum   
2          Cacophis            Cacophis harriettae         harriettae   
3           Litoria        Litoria booroolongensis    booroolongensis   
4       Anthochaera            Anthochaera phrygia            phrygia   
...             ...                            ...                ...   
1017      Fontainea    Fontainea sp. Coffs Harbour  sp. Coffs Harbour   
1018      Melichrus           Melichrus gibberagee         gibberagee   
1019    Pittosporum  Pittosporum sp. Coffs Harbour  sp. Coffs Harbour   
1020      Amytornis       Amytornis striatus howei           striatus   
1021       Cormodes               Cormodes darwini            darwini   

              vernacularName    establishmentMeans           sourceStatus  \
0     Striped Legless Lizard  Alive in NSW, Native             Vulnerable   
1         Gang-gang Cockatoo  Alive in NSW, Native             Vulnerable   
2        White-crowned Snake  Alive in NSW, Native             Vulnerable   
3            Booroolong Frog  Alive in NSW, Native             Endangered   
4          Regent Honeyeater  Alive in NSW, Native  Critically Endangered   
...                      ...                   ...                    ...   
1017                     NaN  Alive in NSW, Native  Critically Endangered   
1018   Narrow-leaf Melichrus  Alive in NSW, Native             Endangered   
1019                     NaN  Alive in NSW, Native  Critically Endangered   
1020                     NaN  Alive in NSW, Native             Endangered   
1021                     NaN  Alive in NSW, Native  Critically Endangered   

      protectedInNSW sensitivityClass  TSProfileID    countryConservation  \
0               True    Not Sensitive      10211.0             Vulnerable   
1               True       Category 3      10975.0             Endangered   
2               True    Not Sensitive      10117.0             Not Listed   
3               True    Not Sensitive      10484.0             Endangered   
4               True    Not Sensitive      10841.0  Critically Endangered   
...              ...              ...          ...                    ...   
1017           False    Not Sensitive      20376.0             Not Listed   
1018           False    Not Sensitive      10521.0             Endangered   
1019           False    Not Sensitive      20378.0             Not Listed   
1020           False    Not Sensitive      20422.0             Not Listed   
1021           False    Not Sensitive      20424.0             Not Listed   

                   dcterms_modified  speciesID  bionetTaxonID  \
0      2013-12-20T15:06:07.13+11:00          1              1   
1      2022-05-12T15:50:01.01+10:00          6              6   
2     2015-05-28T10:14:29.083+10:00         31             31   
3         2011-11-11T11:23:05+11:00         33             33   
4         2011-11-11T11:23:09+11:00         53             53   
...                      

In [91]:
# join them in a way that works for inat (eg sensitive list, geoprivacy = 'obscured'
statelist = pd.concat([sensitivelist[['bionetTaxonID','scientificName','status','bionet_geoprivacy','kingdom','class','family']],
                    conservationlist[['bionetTaxonID','scientificName','status','bionet_geoprivacy','kingdom','class','family']]]).drop_duplicates()
statelist

bionetTaxonID                 scientificName                 status  \
0                 6       Callocephalon fimbriatum             Vulnerable   
1               162                  Ninox strenua             Vulnerable   
2               333               Tyto tenebricosa             Vulnerable   
3               363                Ninox connivens             Vulnerable   
4               390      Hoplocephalus bungaroides             Endangered   
...             ...                            ...                    ...   
1017          24283    Fontainea sp. Coffs Harbour  Critically Endangered   
1018          24284           Melichrus gibberagee             Endangered   
1019          24292  Pittosporum sp. Coffs Harbour  Critically Endangered   
1020          24411       Amytornis striatus howei             Endangered   
1021          24412               Cormodes darwini  Critically Endangered   

     bionet_geoprivacy   kingdom     class          family  
0             obscured  Animalia      Aves      Cacatuidae  
1             obscured  Animalia      Aves       Strigidae  
2             obscured  Animalia      Aves       Tytonidae  
3             obscured  Animalia      Aves       Strigidae  
4             obscured  Animalia  Reptilia        Elapidae  
...                ...       ...       ...             ...  
1017              open   Plantae     Flora   Euphorbiaceae  
1018              open   Plantae     Flora       Ericaceae  
1019              open   Plantae     Flora  Pittosporaceae  
1020              open  Animalia      Aves       Maluridae  
1021              open  Animalia   Insecta        Cleridae  

[1028 rows x 7 columns]

In [92]:
# check for duplicates with conflicting information
statelist.groupby('bionetTaxonID').filter(lambda x: len(x) > 1)#.sort('size',ascending=False)
#df.groupby('hash').filter(lambda group: len(group) > 1).sort('size', ascending=False)

bionetTaxonID    scientificName                 status bionet_geoprivacy  \
144          13940  Wollemia nobilis  Critically Endangered          obscured   
819          13940  Wollemia nobilis  Critically Endangered           private   

     kingdom  class         family  
144  Plantae  Flora  Araucariaceae  
819  Plantae  Flora  Araucariaceae

### 4. Equivalent IUCN statuses

In [93]:
iucn_statuses = {'Not Evaluated', 'Data Deficient', 'Least Concern', 'Near Threatened', 'Vulnerable', 'Endangered', 'Critically Endangered', 'Extinct in the Wild' and 'Extinct'}
sensitivelist.groupby(['status'])['status'].count()

status
Critically Endangered    52
Endangered               93
Extinct                   3
Not Listed                5
Vulnerable               45
Name: status, dtype: int64

In [94]:
iucnStatusMappings = {
    'Not Listed':'Not Evaluated',
}

### 5. Determine best place ID to use

In [95]:
inatstatuses.groupby(['place_id','place_name','place_display_name'])['place_id'].count()
# looks like 6825

place_id  place_name       place_display_name 
6744      Australia        Australia                5
6825      New South Wales  New South Wales, AU    161
Name: place_id, dtype: int64

## Merge iNaturalist statuses with State sensitive list on scientificName

1. Match - updates, even if the statuses are the same we'll update the links and values anyway
2. No match - statuses to be added (additions)
   1.1 No match and no taxnomy - search for synonyms
   1.2 No match
3. Merge the other direction to see if there are deletes?


In [96]:
# join to see which lists already have a status in inaturalist based on scientificName
mergedstatuses = statelist[['bionetTaxonID','scientificName','status','bionet_geoprivacy']].merge(inatstatuses[['status_id','scientificName','taxon_id','user_id','description','iucn','authority','status','geoprivacy','place_id','place_display_name']],how="left",left_on='scientificName',right_on='scientificName',suffixes=(None,'_inat')).sort_values(['scientificName'])
mergedstatuses

bionetTaxonID                     scientificName                 status  \
652           9438                Acacia acanthoclada             Endangered   
443           3132                 Acacia acrionastes             Endangered   
142          13734                       Acacia atrox             Endangered   
512           4746                   Acacia ausfeldii             Vulnerable   
737          11755                      Acacia bakeri             Vulnerable   
..             ...                                ...                    ...   
115          12109                   Zieria parrisiae  Critically Endangered   
798          12110                   Zieria prostrata             Endangered   
799          12111                 Zieria tuberculata             Vulnerable   
819          12277  Zosterops lateralis tephropleurus             Vulnerable   
817          12274                 Zosterops strenuus                Extinct   

    bionet_geoprivacy status_id taxon_id  user_id description iucn  \
652              open       NaN      NaN      NaN         NaN  NaN   
443              open       NaN      NaN      NaN         NaN  NaN   
142          obscured    152295   898643   708886               40   
512              open       NaN      NaN      NaN         NaN  NaN   
737              open       NaN      NaN      NaN         NaN  NaN   
..                ...       ...      ...      ...         ...  ...   
115          obscured       NaN      NaN      NaN         NaN  NaN   
798              open    165911   897289  3669610               40   
799              open       NaN      NaN      NaN         NaN  NaN   
819              open       NaN      NaN      NaN         NaN  NaN   
817              open       NaN      NaN      NaN         NaN  NaN   

                                authority status_inat geoprivacy place_id  \
652                                   NaN         NaN        NaN      NaN   
443                                   NaN         NaN        NaN      NaN   
142  NSW Office of Environment & Heritage  endangered   obscured     6825   
512                                   NaN         NaN        NaN      NaN   
737                                   NaN         NaN        NaN      NaN   
..                                    ...         ...        ...      ...   
115                                   NaN         NaN        NaN      NaN   
798  NSW Office of Environment & Heritage          EN                6825   
799                                   NaN         NaN        NaN      NaN   
819                                   NaN         NaN        NaN      NaN   
817                                   NaN         NaN        NaN      NaN   

      place_display_name  
652                  NaN  
443                  NaN  
142  New South Wales, AU  
512                  NaN  
737                  NaN  
..                   ...  
115                  NaN  
798  New South Wales, AU  
799                  NaN  
819                  NaN  
817                  NaN  

[1031 rows x 14 columns]

In [97]:
# prepare the export fields, common to New template and Update template
# new statuses
# Taxon_Name,Status,Authority,IUCN_equivalent,Description,iNaturalist_Place_ID,url,Taxon_Geoprivacy,Username,taxon_id
# updates
# action,taxon_name,id,taxon_id,status,iucn_equivalent,authority,url,geoprivacy,place_id,username
mergedstatuses['new_authority'] = "New South Wales Office of Environment and Heritage"
mergedstatuses['new_description'] = "Listed as Threatened - refer to https://www.environment.nsw.gov.au/threatenedspeciesapp"
mergedstatuses['new_url'] = "https://www.environment.nsw.gov.au/threatenedspeciesapp/"
# mergedstatuses['new_url'] = "https://www.environment.nsw.gov.au/threatenedspeciesapp/profile.aspx?id=" + mergedstatuses['bionetTaxonID'].astype(str)
mergedstatuses.rename(columns={'bionet_geoprivacy':'new_geoprivacy'},inplace=True)
mergedstatuses['new_place_id'] = '6825'  # NEW SOUTH WALES
mergedstatuses['new_username'] = 'peggydnew'
mergedstatuses['new_iucn_equivalent'] = mergedstatuses['status'].str.lower().str.strip().map(iucnStatusMappings).fillna('Vulnerable') # map to dictionary
mergedstatuses['new_status'] = mergedstatuses['status'].fillna('Threatened')
mergedstatuses

bionetTaxonID                     scientificName                 status  \
652           9438                Acacia acanthoclada             Endangered   
443           3132                 Acacia acrionastes             Endangered   
142          13734                       Acacia atrox             Endangered   
512           4746                   Acacia ausfeldii             Vulnerable   
737          11755                      Acacia bakeri             Vulnerable   
..             ...                                ...                    ...   
115          12109                   Zieria parrisiae  Critically Endangered   
798          12110                   Zieria prostrata             Endangered   
799          12111                 Zieria tuberculata             Vulnerable   
819          12277  Zosterops lateralis tephropleurus             Vulnerable   
817          12274                 Zosterops strenuus                Extinct   

    new_geoprivacy status_id taxon_id  user_id description iucn  \
652           open       NaN      NaN      NaN         NaN  NaN   
443           open       NaN      NaN      NaN         NaN  NaN   
142       obscured    152295   898643   708886               40   
512           open       NaN      NaN      NaN         NaN  NaN   
737           open       NaN      NaN      NaN         NaN  NaN   
..             ...       ...      ...      ...         ...  ...   
115       obscured       NaN      NaN      NaN         NaN  NaN   
798           open    165911   897289  3669610               40   
799           open       NaN      NaN      NaN         NaN  NaN   
819           open       NaN      NaN      NaN         NaN  NaN   
817           open       NaN      NaN      NaN         NaN  NaN   

                                authority  ... geoprivacy place_id  \
652                                   NaN  ...        NaN      NaN   
443                                   NaN  ...        NaN      NaN   
142  NSW Office of Environment & Heritage  ...   obscured     6825   
512                                   NaN  ...        NaN      NaN   
737                                   NaN  ...        NaN      NaN   
..                                    ...  ...        ...      ...   
115                                   NaN  ...        NaN      NaN   
798  NSW Office of Environment & Heritage  ...                6825   
799                                   NaN  ...        NaN      NaN   
819                                   NaN  ...        NaN      NaN   
817                                   NaN  ...        NaN      NaN   

      place_display_name                                      new_authority  \
652                  NaN  New South Wales Office of Environment and Heri...   
443                  NaN  New South Wales Office of Environment and Heri...   
142  New South Wales, AU  New South Wales Office of Environment and Heri...   
512                  NaN  New South Wales Office of Environment and Heri...   
737                  NaN  New South Wales Office of Environment and Heri...   
..                   ...                                                ...   
115                  NaN  New South Wales Office of Environment and Heri...   
798  New South Wales, AU  New South Wales Office of Environment and Heri...   
799                  NaN  New South Wales Office of Environment and Heri...   
819                  NaN  New South Wales Office of Environment and Heri...   
817                  NaN  New South Wales Office of Environment and Heri...   

                                       new_description  \
652  Listed as Threatened - refer to https://www.en...   
443  Listed as Threatened - refer to https://www.en...   
142  Listed as Threatened - refer to https://www.en...   
512  Listed as Threatened - refer to https://www.en...   
737  Listed as Threatened - refer to https://www.en...   
..                                                 ...   
115  Listed as Threatened - refer to https:

## Updates

In [98]:
# updates
# action,taxon_name,id,taxon_id,status,iucn_equivalent,authority,url,geoprivacy,place_id,username,description
updates = pd.DataFrame(mergedstatuses[mergedstatuses['status_id'].notnull()])
updates.sort_values('scientificName')
updates['action'] = 'UPDATE'
#updates.loc[:,'action'] = 'UPDATE'
updates = updates[['action','scientificName','status_id','taxon_id','new_status','new_iucn_equivalent','new_authority','new_url','new_geoprivacy','new_place_id','new_username','new_description']]
updates.columns = updates.columns.str.replace("new_", "", regex=True)
updates = updates.rename(columns={'scientificName':'taxon_name',
                                  'status_id':'id'})
updates

action             taxon_name      id taxon_id                 status  \
142  UPDATE           Acacia atrox  152295   898643             Endangered   
627  UPDATE   Acacia baueri aspera  159920  1111698             Vulnerable   
97   UPDATE     Acacia dangarensis  152286   775137  Critically Endangered   
612  UPDATE       Acacia pubifolia  162893   897779             Endangered   
449  UPDATE          Acacia ruppii  162887   578518             Endangered   
..      ...                    ...     ...      ...                    ...   
112  UPDATE  Viola cleistogamoides  152321   566603             Endangered   
145  UPDATE       Wollemia nobilis  152312    49381  Critically Endangered   
879  UPDATE       Wollemia nobilis  152312    49381  Critically Endangered   
708  UPDATE     Zieria involucrata  266172   604220             Endangered   
798  UPDATE       Zieria prostrata  165911   897289             Endangered   

    iucn_equivalent                                          authority  \
142      Vulnerable  New South Wales Office of Environment and Heri...   
627      Vulnerable  New South Wales Office of Environment and Heri...   
97       Vulnerable  New South Wales Office of Environment and Heri...   
612      Vulnerable  New South Wales Office of Environment and Heri...   
449      Vulnerable  New South Wales Office of Environment and Heri...   
..              ...                                                ...   
112      Vulnerable  New South Wales Office of Environment and Heri...   
145      Vulnerable  New South Wales Office of Environment and Heri...   
879      Vulnerable  New South Wales Office of Environment and Heri...   
708      Vulnerable  New South Wales Office of Environment and Heri...   
798      Vulnerable  New South Wales Office of Environment and Heri...   

                                                   url geoprivacy place_id  \
142  https://www.environment.nsw.gov.au/threateneds...   obscured     6825   
627  https://www.environment.nsw.gov.au/threateneds...       open     6825   
97   https://www.environment.nsw.gov.au/threateneds...   obscured     6825   
612  https://www.environment.nsw.gov.au/threateneds...       open     6825   
449  https://www.environment.nsw.gov.au/threateneds...       open     6825   
..                                                 ...        ...      ...   
112  https://www.environment.nsw.gov.au/threateneds...   obscured     6825   
145  https://www.environment.nsw.gov.au/threateneds...   obscured     6825   
879  https://www.environment.nsw.gov.au/threateneds...    private     6825   
708  https://www.environment.nsw.gov.au/threateneds...       open     6825   
798  https://www.environment.nsw.gov.au/threateneds...       open     6825   

      username                                        description  
142  peggydnew  Listed as Threatened - refer to https://www.en...  
627  peggydnew  Listed as Threatened - refer to https://www.en...  
97   peggydnew  Listed as Threatened - refer to https://www.en...  
612  peggydnew  Listed as Threatened - refer to https://www.en...  
449  peggydnew  Listed as Threatened - refer to https://www.en...  
..         ...                                                ...  
112  peggydnew  Listed as Threatened - refer to https://www.en...  
145  peggydnew  Listed as Threatened - refer to https://www.en...  
879  peggydnew  Listed as Threatened - refer to https://www.en...  
708  peggydnew  Listed as Threatened - refer to https://www.en...  
798  peggydnew  Listed as Threatened - refer to https://www.en...  

[150 rows x 12 columns]

## No status in iNaturalist via straight scientificName match
The NSW records that didn't match up to a status in iNaturalist

In [99]:
# to add: those that have no inaturalist status - 532!!
noinatstatus = mergedstatuses[mergedstatuses['status_id'].isnull()]
# try to match the taxon name to something in inaturalist
noinatstatus = noinatstatus.merge(inattaxa, how="left", left_on="scientificName",right_on="scientificName")
noinatstatus

bionetTaxonID                     scientificName                 status  \
0             9438                Acacia acanthoclada             Endangered   
1             3132                 Acacia acrionastes             Endangered   
2             4746                   Acacia ausfeldii             Vulnerable   
3            11755                      Acacia bakeri             Vulnerable   
4             4964                    Acacia bynoeana             Endangered   
..             ...                                ...                    ...   
876          22908     Zieria odorifera warrabahensis  Critically Endangered   
877          12109                   Zieria parrisiae  Critically Endangered   
878          12111                 Zieria tuberculata             Vulnerable   
879          12277  Zosterops lateralis tephropleurus             Vulnerable   
880          12274                 Zosterops strenuus                Extinct   

    new_geoprivacy status_id taxon_id user_id description iucn authority  ...  \
0             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
1             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
2             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
3             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
4             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
..             ...       ...      ...     ...         ...  ...       ...  ...   
876           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
877       obscured       NaN      NaN     NaN         NaN  NaN       NaN  ...   
878           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
879           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
880           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   

           phylum          class          order        family      genus  \
0    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
1    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
2    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
3    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
4    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
..            ...            ...            ...           ...        ...   
876  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
877  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
878  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
879      Chordata           Aves  Passeriformes  Zosteropidae  Zosterops   
880      Chordata           Aves  Passeriformes  Zosteropidae  Zosterops   

    specificEpithet infraspecificEpithet              modified   taxonRank  \
0      acanthoclada                  NaN  2022-08-22T22:51:08Z     species   
1       acrionastes                  NaN  2022-04-06T22:12:50Z     species   
2         ausfeldii                  NaN  2022-06-07T09:01:00Z     species   
3            bakeri                  NaN  2022-04-07T01:35:47Z     species   
4          bynoeana                  NaN  2022-04-07T01:36:15Z     species   
..              ...                  ...                   ...         ...   
876       odorifera        warrabahensis  2021-07-28T04:47:59Z  subspecies   
877       parrisiae                  NaN  2020-02-19T19:35:12Z     species   
878     tuberculata                  NaN  2021-07-28T04:47:52Z     species   
879       lateralis        tephropleurus  2020-01-16T00:05:01Z  subspecies   
880        strenuus                  NaN  2019-11-23T02:19:57Z     species   

                                            references  
0    http://www.catalogueoflife.org/annual-checklis...  
1                         http://powo.science.kew.org/  
2    http://www.catalogueoflife.org/annual-checkli

In [100]:
additions = pd.DataFrame(noinatstatus[noinatstatus['id'].notna()])
additions

bionetTaxonID                     scientificName                 status  \
0             9438                Acacia acanthoclada             Endangered   
1             3132                 Acacia acrionastes             Endangered   
2             4746                   Acacia ausfeldii             Vulnerable   
3            11755                      Acacia bakeri             Vulnerable   
4             4964                    Acacia bynoeana             Endangered   
..             ...                                ...                    ...   
876          22908     Zieria odorifera warrabahensis  Critically Endangered   
877          12109                   Zieria parrisiae  Critically Endangered   
878          12111                 Zieria tuberculata             Vulnerable   
879          12277  Zosterops lateralis tephropleurus             Vulnerable   
880          12274                 Zosterops strenuus                Extinct   

    new_geoprivacy status_id taxon_id user_id description iucn authority  ...  \
0             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
1             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
2             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
3             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
4             open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
..             ...       ...      ...     ...         ...  ...       ...  ...   
876           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
877       obscured       NaN      NaN     NaN         NaN  NaN       NaN  ...   
878           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
879           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   
880           open       NaN      NaN     NaN         NaN  NaN       NaN  ...   

           phylum          class          order        family      genus  \
0    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
1    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
2    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
3    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
4    Tracheophyta  Magnoliopsida        Fabales      Fabaceae     Acacia   
..            ...            ...            ...           ...        ...   
876  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
877  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
878  Tracheophyta  Magnoliopsida     Sapindales      Rutaceae     Zieria   
879      Chordata           Aves  Passeriformes  Zosteropidae  Zosterops   
880      Chordata           Aves  Passeriformes  Zosteropidae  Zosterops   

    specificEpithet infraspecificEpithet              modified   taxonRank  \
0      acanthoclada                  NaN  2022-08-22T22:51:08Z     species   
1       acrionastes                  NaN  2022-04-06T22:12:50Z     species   
2         ausfeldii                  NaN  2022-06-07T09:01:00Z     species   
3            bakeri                  NaN  2022-04-07T01:35:47Z     species   
4          bynoeana                  NaN  2022-04-07T01:36:15Z     species   
..              ...                  ...                   ...         ...   
876       odorifera        warrabahensis  2021-07-28T04:47:59Z  subspecies   
877       parrisiae                  NaN  2020-02-19T19:35:12Z     species   
878     tuberculata                  NaN  2021-07-28T04:47:52Z     species   
879       lateralis        tephropleurus  2020-01-16T00:05:01Z  subspecies   
880        strenuus                  NaN  2019-11-23T02:19:57Z     species   

                                            references  
0    http://www.catalogueoflife.org/annual-checklis...  
1                         http://powo.science.kew.org/  
2    http://www.catalogueoflife.org/annual-checkli

In [101]:
# there's no status but there is a matching inat taxon (id is the taxon id)
additions = pd.DataFrame(noinatstatus[noinatstatus['id'].notna()])
additions.sort_values(['scientificName'])
additions['action'] = 'ADD'
additions = additions[['action','scientificName','status_id','id','new_status','new_iucn_equivalent','new_authority','new_url','new_geoprivacy','new_place_id','new_username','new_description']]
additions.columns = additions.columns.str.replace("new_", "", regex=True)
additions = additions.rename(columns={'scientificName':'taxon_name',
                                      'id':'taxon_id',
                                  'status_id':'id'})
additions

action                         taxon_name   id taxon_id  \
0      ADD                Acacia acanthoclada  NaN   787701   
1      ADD                 Acacia acrionastes  NaN   898576   
2      ADD                   Acacia ausfeldii  NaN   139906   
3      ADD                      Acacia bakeri  NaN   524410   
4      ADD                    Acacia bynoeana  NaN   543213   
..     ...                                ...  ...      ...   
876    ADD     Zieria odorifera warrabahensis  NaN  1247632   
877    ADD                   Zieria parrisiae  NaN   965722   
878    ADD                 Zieria tuberculata  NaN  1247618   
879    ADD  Zosterops lateralis tephropleurus  NaN   980046   
880    ADD                 Zosterops strenuus  NaN    17457   

                    status iucn_equivalent  \
0               Endangered      Vulnerable   
1               Endangered      Vulnerable   
2               Vulnerable      Vulnerable   
3               Vulnerable      Vulnerable   
4               Endangered      Vulnerable   
..                     ...             ...   
876  Critically Endangered      Vulnerable   
877  Critically Endangered      Vulnerable   
878             Vulnerable      Vulnerable   
879             Vulnerable      Vulnerable   
880                Extinct      Vulnerable   

                                             authority  \
0    New South Wales Office of Environment and Heri...   
1    New South Wales Office of Environment and Heri...   
2    New South Wales Office of Environment and Heri...   
3    New South Wales Office of Environment and Heri...   
4    New South Wales Office of Environment and Heri...   
..                                                 ...   
876  New South Wales Office of Environment and Heri...   
877  New South Wales Office of Environment and Heri...   
878  New South Wales Office of Environment and Heri...   
879  New South Wales Office of Environment and Heri...   
880  New South Wales Office of Environment and Heri...   

                                                   url geoprivacy place_id  \
0    https://www.environment.nsw.gov.au/threateneds...       open     6825   
1    https://www.environment.nsw.gov.au/threateneds...       open     6825   
2    https://www.environment.nsw.gov.au/threateneds...       open     6825   
3    https://www.environment.nsw.gov.au/threateneds...       open     6825   
4    https://www.environment.nsw.gov.au/threateneds...       open     6825   
..                                                 ...        ...      ...   
876  https://www.environment.nsw.gov.au/threateneds...       open     6825   
877  https://www.environment.nsw.gov.au/threateneds...   obscured     6825   
878  https://www.environment.nsw.gov.au/threateneds...       open     6825   
879  https://www.environment.nsw.gov.au/threateneds...       open     6825   
880  https://www.environment.nsw.gov.au/threateneds...       open     6825   

      username                                        description  
0    peggydnew  Listed as Threatened - refer to https://www.en...  
1    peggydnew  Listed as Threatened - refer to https://www.en...  
2    peggydnew  Listed as Threatened - refer to https://www.en...  
3    peggydnew  Listed as Threatened - refer to https://www.en...  
4    peggydnew  Listed as Threatened - refer to https://www.en...  
..         ...                                                ...  
876  peggydnew  Listed as Threatened - refer to https://www.en...  
877  peggydnew  Listed as Threatened - refer to https://www.en...  
878  peggydnew  Listed as Threatened - refer to https://www.en...  
879  peggydnew  Listed as Threatened - refer to https://www.en...  
880  peggydnew  Listed as Threatened - refer to https://www.en...  

[732 rows x 12 columns]

In [102]:
# write these to the file
pd.concat([updates,additions]).to_csv(sourcedir + "nsw.csv", index=False)

In [103]:
# what didnt match to a taxon?
unknownToInat = noinatstatus[noinatstatus['id'].isna()]
unknownToInat

bionetTaxonID                       scientificName  \
23           24269     Acacia terminalis Eastern Sydney   
37           12048  Amperea xiphoclada var. pedicellata   
43           24411             Amytornis striatus howei   
44           14551          Amytornis striatus striatus   
46           12355                       Angophora exul   
..             ...                                  ...   
831          10302              Trianthema cypseleoides   
837          12096                   Tylophora linearis   
838           2396                   Tylophora woollsii   
842          13469           Typhonium sp. aff. brownii   
847           8956                   Velleia perfoliata   

                    status new_geoprivacy status_id taxon_id user_id  \
23              Endangered           open       NaN      NaN     NaN   
37                 Extinct           open       NaN      NaN     NaN   
43              Endangered           open       NaN      NaN     NaN   
44   Critically Endangered           open       NaN      NaN     NaN   
46              Endangered       obscured       NaN      NaN     NaN   
..                     ...            ...       ...      ...     ...   
831                Extinct           open       NaN      NaN     NaN   
837             Vulnerable           open       NaN      NaN     NaN   
838             Endangered           open       NaN      NaN     NaN   
842             Endangered       obscured       NaN      NaN     NaN   
847             Vulnerable           open       NaN      NaN     NaN   

    description iucn authority  ... phylum class order family genus  \
23          NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
37          NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
43          NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
44          NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
46          NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
..          ...  ...       ...  ...    ...   ...   ...    ...   ...   
831         NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
837         NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
838         NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
842         NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   
847         NaN  NaN       NaN  ...    NaN   NaN   NaN    NaN   NaN   

    specificEpithet infraspecificEpithet modified taxonRank references  
23              NaN                  NaN      NaN       NaN        NaN  
37              NaN                  NaN      NaN       NaN        NaN  
43              NaN                  NaN      NaN       NaN        NaN  
44              NaN                  NaN      NaN       NaN        NaN  
46              NaN                  NaN      NaN       NaN        NaN  
..              ...                  ...      ...       ...        ...  
831             NaN                  NaN      NaN       NaN        NaN  
837             NaN                  NaN      NaN       NaN        NaN  
838             NaN                  NaN      NaN       NaN        NaN  
842             NaN                  NaN      NaN       NaN        NaN  
847             NaN                  NaN      NaN       NaN        NaN  

[149 rows x 36 columns]

### are there any that need to be removed?
nsw sensitive list count: 198
nsw inat statuses count: 166

updates to inat status: 150
additional inat status: 732
nsw statuses we can't find a taxon match for in iNaturalist: 149
total: 1395 (explainable via the various genus/section entries that we matched to in the taxonomy)

inat statuses left over: 166-150=16 that may need checking

In [104]:
# inat statuses that aren't in added or updated
inatstatuses[~inatstatuses['taxon_id'].isin(updates['taxon_id'])]


status_id taxon_id  user_id place_id source_id  \
477     264941  1061113  3669610     6825             
681     164757  1098159    58320     6825             
1147    152307   116844   708886     6825     16650   
736     166035  1194478  3669610     6825             
737     166036  1194478  3669610     6744             
51      167249  1227491  3669610     6825             
397     264070  1255425  3669610     6825             
809     262163  1289564   708886     6825     16650   
3367    152252    19111   708886     6825     16650   
2653    164637    19275    84719     6825             
2608    164300   369148   516268     6825             
1130    152282   505120   708886     6825     16650   
65      168033   634763   702203     6825             
1137    152293   775274   708886     6825     16650   
3426    152239   792286   708886     6825     16650   
2437    165525   906591    58320     6825             
836     155144   966856   708886     6825     16650   

                                              authority  \
477   New South Wales Biodiversity Conservation Act ...   
681                NSW Office of Environment & Heritage   
1147               NSW Office of Environment & Heritage   
736                NSW Office of Environment & Heritage   
737                               Australian Government   
51                 NSW Office of Environment & Heritage   
397                NSW Office of Environment & Heritage   
809                NSW Office of Environment & Heritage   
3367               NSW Office of Environment & Heritage   
2653                                      IUCN Red List   
2608        NSW Threatened Species Scientific Committee   
1130               NSW Office of Environment & Heritage   
65               Threatened Species Protection Act 1995   
1137               NSW Office of Environment & Heritage   
3426               NSW Office of Environment & Heritage   
2437               NSW Office of Environment & Heritage   
836                NSW Office of Environment & Heritage   

                     status  \
477                      EX   
681              Vulnerable   
1147  critically endangered   
736                      EN   
737                      EN   
51                       EN   
397                      CR   
809              vulnerable   
3367             vulnerable   
2653                     EN   
2608                     EN   
1130             vulnerable   
65                       EN   
1137  endangered population   
3426             vulnerable   
2437             Endangered   
836              endangered   

                                                    url  \
477   https://bie.ala.org.au/species/https://id.biod...   
681       https://www.inaturalist.org/taxa/1098159/edit   
1147  https://www.environment.nsw.gov.au/resources/a...   
736   https://www.environment.nsw.gov.au/threateneds...   
737   https://www.environment.nsw.gov.au/threateneds...   
51    https://www.environment.nsw.gov.au/threateneds...   
397   https://www.environment.nsw.gov.au/threateneds...   
809   https://www.environment.nsw.gov.au/resources/a...   
3367  https://www.environment.nsw.gov.au/resources/a...   
2653  https://www.iucnredlist.org/species/22727593/1...   
2608  https://www.dpi.nsw.gov.au/__data/assets/pdf_f...   
1130  https://www.environment.nsw.gov.au/resources/a...   
65    http://www.environment.gov.au/resource/nationa...   
1137  https://www.environment.nsw.gov.au/resources/a...   
3426  https://www.environment.nsw.gov.au/resources/a...   
2437  https://www.environment.nsw.gov.au/threatenedS...   
836   https://www.environment.nsw.gov.au/resources/a...   

                                            description geoprivacy  ...  \
477                                    Presumed Extinct       open  ...   
681                                                                 ...   
1147                                                      obscured  ...   
736                          